app/static/js/initialize_map.js

```diff
@@ -0,0 +1,285 @@
+    $("#slider").dateRangeSlider("max", BOUNDS_MAX);
+}
+
+//Use Google geocoder to update geolocation given an address through
+//the address search box
+function update_center() {
+    geocoder = new google.maps.Geocoder();
+    address = $("#address").val();
+    if (geocoder) {
+        console.log(address);
+        geocoder.geocode( { 'address': address},
+        function(results, status) {
+            if (status != google.maps.GeocoderStatus.OK) {
+                alert("Geocode not successful - " + status);
+            }
+            else if (status == google.maps.GeocoderStatus.ZERO_RESULTS) {
+                alert("Address not found!");
+            }
+            else {
+                map.setCenter(results[0].geometry.location);
+            }
+        });
+    }
+}
+
+//Get address submit event
+$(document).ready(function() {
+    $('#addressForm').on('submit', function (event) {
+        update_center();
+        return false;
+    });
+});
+
+//Date Slider - TODO
+$(function() {
+    $("#slider").dateRangeSlider({
+        bounds: {
+            min: BOUNDS_MIN,
+            max: BOUNDS_MAX,
+        },
+        arrows: false,
+        defaultValues: {
+            min: BOUNDS_MIN,
+            max: BOUNDS_MAX,
+        },
+    });
+    $("#slider").bind("valuesChanged", function(e, data) {
+        console.log("Values just changed. min: " + String(data.values.min).substring(4, 15) + " max: " + String(data.values.max).substring(4, 15));
+        var beginYear = parseInt(String(data.values.min).substring(11, 15), 10);
+        var beginDay = parseInt(String(data.values.min).substring(8, 10), 10);
+        var endYear = parseInt(String(data.values.max).substring(11, 15), 10);
+        var endDay = parseInt(String(data.values.min).substring(8, 10), 10);
+        var beginMonth = 0;
+        var endMonth = 0;
+        switch(String(data.values.min).substring(4, 7)) {
+            case "Jan":
+                beginMonth = 0;
+                break;
+            case "Feb":
+                beginMonth = 1;
+                break;
+            case "Mar":
+                beginMonth = 2;
+                break;
+            case "Apr":
+                beginMonth = 3;
+                break;
+            case "May":
+                beginMonth = 4;
+                break;
+            case "Jun":
+                beginMonth = 5;
+                break;
+            case "Jul":
+                beginMonth = 6;
+                break;
+            case "Aug":
+                beginMonth = 7;
+                break;
+            case "Sep":
+                beginMonth = 8;
+                break;
+            case "Oct":
+                beginMonth = 9;
+                break;
+            case "Nov":
+                beginMonth = 10;
+                break;
+            case "Dec":
+                beginMonth = 11;
+                break;
+        }
+        switch(String(data.values.max).substring(4, 7)) {
```

Comment:
If you were to keep using a switch statement you should refactor this repeated code into its own function